### PR TITLE
Lcg/suppress reporter failures

### LINKED ISSUE #4302: Resource reporter throws Exception when filename contains ISO-8859-1 encoded characters
Hi,
I am trying to add a file resource, where the file name has french characters. I guess below error in the log and chef client is stopped abruptly.

Chef::Handler::JsonFile raised #<Chef::Exceptions::JSON::EncodeError: Invalid UTF-8 string 'Encoding::InvalidByteSequenceError

Encoding::InvalidByteSequenceError: "\xE9" followed by "s" on UTF-8': cannot encode to UTF-8

Can you provide some suggessions.

### PR DESCRIPTION
**Pull request by lamont-granquist**
fixes #4302

### CODE REVIEW

**Comment by lamont-granquist** [comment]
working on specs...

**Comment by jaym** [comment]
:+1:

**Comment by thommay** [comment]
yep :+1:

File: lib/chef/client.rb [hunk 1]
```diff
@@ -603,6 +621,8 @@ def register(client_name=node_name, config=Chef::Config)
       # We now have the client key, and should use it from now on.
       @rest = Chef::ServerAPI.new(config[:chef_server_url], client_name: client_name,
                                   signing_key_filename: config[:client_key])
+      # force initialization of ther rest_clean API object
```
Comment (thommay) [inline lib/chef/client.rb]: "the"

**Pull request merged**